File: python/character_match.py
# Don't forget to run your tests!

from operator import is_


def is_character_match(string1, string2):

	string1, string2 = string1.lower(), string2.lower()
	string1, string2 = string1.replace(" ", ""), string2.replace(" ", "")

	
	x = {n: string1.lower().replace(" ", "").count(n) for n in string1.replace(" ", "").lower()}
	y = {n: string2.lower().replace(" ", "").count(n) for n in string2.replace(" ", "").lower()}

	return x == y

	

	# solved in one line
	# return {n: string1.lower().replace(" ", "").count(n) for n in string1.replace(" ", "").lower()} == {n: string2.lower().replace(" ", "").count(n) for n in string2.replace(" ", "").lower()}

File: js/characterMatch.js
exports.isCharacterMatch = function(string1, string2) {

    string1 = string1.toLowerCase().replace(/ /g, "")
    string2 = string2.toLowerCase().replace(/ /g, "")

    string1 = string1.split("")
    string2 = string2.split("")

    string1 = string1.sort()
    string2 = string2.sort()

    string1 = string1.join("")
    string2 = string2.join("")

    return string1 === string2
};


